chalk-solve/src/wf.rs

```diff
@@ -26,6 +27,7 @@ pub enum WfError {
 #[derive(new)]
 pub struct WfSolver<'db, DB: ChalkSolveDatabase> {
     db: &'db DB,
+    solver_choice: SolverChoice,
```

Comment:
Maybe we should make this store the *solver*, not the solver-choice. This would allow for more caching (basically, we would invoke `into_solver` from within the `new` method)

Reply:
I was wondering why [the commit that originally removed this](https://github.com/rust-lang/chalk/commit/53614c9dbdb5546161007c0446fe59f2b3b39a5e#diff-a48af376562b9620a110c2c78c1da2bdL30) did this, but it turns out that doing `self.solver.solve()` would require the solver to be mutable, which was avoided by creating a new one every time.

Changing the functions to take `&mut self` would probably pollute the code a bit.

Reply:
OK, it's fine, we can leave it as is for now.